app/__init__.py

```diff
@@ -34,106 +40,197 @@
+    """
     try:
-        data = request.json  # Get request body
-        name = data["name"]
+        workspace = get_connection().db_search(name)
         # path specifying location of working directory inside server
-        for x in os.listdir(os.path.join(app.root_path + WORKING_DIR)):
-            if x == name:
-                shutil.rmtree(name)
-                return jsonify(name=name,
-                               status="Workspace deleted successfully")
-        return jsonify(status="Workspace " + name + " not found")
+        return Response(json.dumps(workspace), status=200,
+                        mimetype='application/json')
     except Exception as e:
         app.logger.error(e)
-        return jsonify(status=409, message=str(e))
+        return jsonify(status=errors.ERROR_STATUS, message=str(e))
 
 
-@app.route('/workspace/fetch', methods=['POST'])
-def linchpin_fetch_workspace():
+@app.route('/workspace/delete/<identity>', methods=['DELETE'])
+def linchpin_delete_workspace(identity) -> Response:
+    """
+        DELETE request route for deleting workspaces.
+        RequestBody: {"name": "workspacename"}
+        :return : response with deleted workspace name and status
+    """
     try:
-        data = request.json  # Get request body
-        name = data['name']
-        url = data['url']
-        repo = None
-        # initial list
-        cmd = ["linchpin", "-w " + WORKING_DIR + name + "/", "fetch"]
-
-        # Check for repoType field in request,
-        # Only true if it is set to web
-        if 'repoType' in data:
-            if data['repoType'] == 'web':
-                repo = 'web'
-                cmd.append("--web")
-
-        if 'rootfolder' in data:
-            cmd.extend(("--root", data['rootfolder']))
-
-        if repo is None and 'branch' in data:
-            cmd.extend(("--branch", data['branch']))
-
-        # last item to be added in the array
-        if 'url' in data:
-            cmd.append(str(url))
-        # Checking if workspace already exists
-        if os.path.exists(os.path.join(app.root_path,
-                                       WORKING_DIR + "/" + name)):
-            return jsonify(status="workspace with the same "
-                                  "name found try again by renaming")
-        else:
-            output = subprocess.Popen(cmd, stdout=subprocess.PIPE)
-            if check_workspace_empty(name):
-                return jsonify(message="Only public repositories can be "
-                                       "used as fetch URl's")
-            return jsonify(name=data["name"], status="Workspace created "
-                                                     "successfully",
-                           code=output.returncode)
+        # path specifying location of working directory inside server
+        for x in os.listdir(WORKING_PATH):
+            if x == identity:
+                shutil.rmtree(WORKING_PATH + "/" + x)
```

Comment:
@mansikulkarni96 
what does identity mean ? uuid + workspace name or just uuid ? 
If its uuid_workspacename it works.

Reply:
identity throughout this application is uuid_workspacename, I couldn't use name "id" as it shadows a built-in name.